src/vehicle.cpp

```diff
@@ -6849,64 +6841,88 @@ long vehicle_part::ammo_capacity() const
 {
     // @todo currently only support fuel tanks and batteries
 
-    if( base.is_magazine() ) {
+    if( base.is_magazine() || base.typeId() == "minireactor" ) {
         return base.ammo_capacity();
     }
 
-    return info().has_flag( VPFLAG_FUEL_TANK ) ? info().size : 0;
+    if( base.is_watertight_container() ) {
+        return base.type->container->contains * std::max( item::find_type( ammo_current() )->stack_size, 1 );
```

Comment:
I'm not sure if the code for `item::liquid_charges` is correct - it depends on `itype::*::def_charges` which is not necessarily equal to `stack_size`. The latter should determine total volume and not the former.

Changing `liquid_charges` isn't feasible as much code probably accidentally depends upon this.